### PR TITLE
ROX-18903: Generate reporting v2 api docs

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Description

- Updated Makefiles and dockerfile to support API docs generation for desired v2 APIs.
- Generated v2 report_service API docs

## Checklist
- [ ] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Manual testing to see if docs are exposed in UI

image

image

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by md2119 — CHANGES_REQUESTED** [review]
We should avoid merging the API documentation of different versions. 
I recommend the following:
- Generate the API documentation of `v2` APIs at a location different from the `v1` documentation location
- A separate API for the `v2` docs
- A separate UI menu option for `v2` docs

File: central/main.go
Comment (md2119) [inline central/main.go:684]: The existing API should be handled by the existing handler.

File: central/main.go
Comment (md2119) [inline central/main.go:690]: `docs.SwaggerV2()`

**Review by md2119 — CHANGES_REQUESTED** [review]

File: central/main.go
Comment (charmik-redhat) [reply central/main.go]: Ah I messed up ordering. Fixed it.

**Comment by vjwilson** [comment]
@charmik-redhat I don't think having two links in the nav is best way to handle this, but if we are trying to get this in 4.3, then we can launch v2 docs like that.

I think it would be better if the UI team built our own API version "switcher".

Do you need this in 4.3, or could it be merged sometime after the code freeze?

cc: @md2119 @stackrox/ui-dep-updaters

**Comment by md2119** [comment]
> @charmik-redhat I don't think having two links in the nav is best way to handle this, but if we are trying to get this in 4.3, then we can launch v2 docs like that.
> 
> I think it would be better if the UI team built our own API version "switcher".
> 
> Do you need this in 4.3, or could it be merged sometime after the code freeze?
> 
> cc: @md2119 @stackrox/ui-dep-updaters

It would be good to have in-product API docs given since the reporting feature will be generally available in 4.3. The crux is that we do not want to merge API docs of different versions. @vjwilson, I believe you are hinting at a better UX that would also adhere to this principle. 

My recommendation in this case would be to evaluate the risk of the UI change. If Charmik's proposal has low risk, go with his proposal in 4.3 and iterate over the UX in a later release.

**Comment by charmik-redhat** [comment]
> > @charmik-redhat I don't think having two links in the nav is best way to handle this, but if we are trying to get this in 4.3, then we can launch v2 docs like that.
> > I think it would be better if the UI team built our own API version "switcher".
> > Do you need this in 4.3, or could it be merged sometime after the code freeze?
> > cc: @md2119 @stackrox/ui-dep-updaters
> 
> It would be good to have in-product API docs given since the reporting feature will be generally available in 4.3. The crux is that we do not want to merge API docs of different versions. @vjwilson, I believe you are hinting at a better UX that would also adhere to this principle.
> 
> My recommendation in this case would be to evaluate the risk of the UI change. If Charmik's proposal has low risk, go with his proposal in 4.3 and iterate over the UX in a later release.

Agree with this.

File: ui/apps/platform/src/routePaths.ts
Comment (vjwilson) [inline ui/apps/platform/src/routePaths.ts:430]: @charmik-redhat 
Thanks for adding the screenshot.

I think we should explicit label the link for the existing v1 API as "(v1)" in the menu link. (and also have a space between the name and version)

```suggestion
    [apidocsPath]: 'API Reference (v2)',
    [apidocsPathV2]: 'API Reference (v2)',
```

Comment (vjwilson) [reply ui/apps/platform/src/routePaths.ts:430]: But leave the original URL path the same, because we will go back to one URL when we put a version selector to collapse it back to one page.

**Review by md2119 — APPROVED** [review]
The backend changes look good. Wait for UI team's approval for the UI changes before merging.

File: ui/apps/platform/src/routePaths.ts
Comment (charmik-redhat) [reply ui/apps/platform/src/routePaths.ts]: Thanks, that sounds better. Updated the text to `API Reference (v1)` and `API Reference (v2)`

**Review by vjwilson — APPROVED** [review]
UI looks good now

**Pull request merged**
